### PR TITLE
Encrypt generated key with AES

### LINKED ISSUE #50843: Encrypt PEM files with AES
`elasticsearch-certutil` uses DES-EDE3-CBC for encrypting private keys when writing them to PEM files.

It should be considered to move to an AES based one.

### PR DESCRIPTION
**Pull request by ywangd**
Replace DES with AES to align with modern encryption standards

Resolves: #50843

### CODE REVIEW

**Comment by tvernum** [comment]
@ywangd Can you modify the title of the PR to refer to `key` rather than `cert`?
Thanks.

**Comment by ywangd** [comment]
Failure is genuine. Because of randomness, sometimes the key file does not have any encryption. Will fix it.

**Comment by tvernum** [comment]
@ywangd Did you accidentally close this?

**Comment by ywangd** [comment]
Must be me then. Sorry reopenning

File: x-pack/plugin/security/cli/src/test/java/org/elasticsearch/xpack/security/cli/CertificateToolTests.java [hunk 1]
```diff
@@ -368,7 +380,11 @@ public void testGeneratingSignedPemCertificates() throws Exception {
             assertTrue(Files.exists(zipRoot.resolve(filename)));
             final Path cert = zipRoot.resolve(filename + "/" + filename + ".crt");
             assertTrue(Files.exists(cert));
-            assertTrue(Files.exists(zipRoot.resolve(filename + "/" + filename + ".key")));
+            Path keyFile = zipRoot.resolve(filename + "/" + filename + ".key");
+            assertTrue(Files.exists(keyFile));
+            if (keyPassword != null) {
+                assertTrue(Files.readString(keyFile).contains("DEK-Info: AES-128-CBC"));
+            }
```
Comment (tvernum) [inline x-pack/plugin/security/cli/src/test/java/org/elasticsearch/xpack/security/cli/CertificateToolTests.java]: For completeness, can we also
```suggestion
            } else {
               assertFalse(Files.readString(keyFile).contains("DEK-Info:"));
            }
```

**Review by jkakavas — APPROVED** [review]
LGTM

**Review by tvernum — APPROVED** [review]
LGTM

**Comment by ywangd** [comment]
Done backport to 7.x

**Pull request merged**